src/openrct2/peep/peep.c

```diff
@@ -10108,7 +10109,7 @@ static sint32 guest_path_finding(rct_peep* peep)
 	z = peep->next_z;
 
 	rct_map_element *mapElement = map_get_path_element_at(x / 32, y / 32, z);
-	if (mapElement == NULL) {
+	if (mapElement == NULL || (mapElement->flags & MAP_ELEMENT_FLAG_GHOST)) {
```

Comment:
It might be better to have a version of map_get_path_element_at or modify that to ignore ghosts. Ghosts can easily have the same x, y and z as an existing path element (with disable clearance checks on).